File: Vertask - Ionic/src/app/model/tarefa.ts
import {Usuario} from './usuario';
import { StatusTarefa } from '../enum/status-tarefa.enum';
export class Tarefa {

    idTarefa: number;
    nome: string;
    descricao: string;
    dataInicio: Date;
    dataEntrega: Date;
    statusTarefa: StatusTarefa;
    idAdministrador: number;
    favorita: boolean;
    // Se true, o administrador receberá notificações por e-mail sobre início/conclusão
    notifyAdmin: boolean;
    // Comentário/observação informada ao concluir a tarefa
    observacao?: string;

    usuarios: Usuario[];
    usuariosIds: number[]; // Array para armazenar IDs dos usuários associados
    clienteId?: number;
    
    // Metadados do documento (anexo)
    documentoNome?: string;
    documentoMime?: string;
    documentoTamanho?: number;
    documentoPath?: string;


    constructor() {
        this.idTarefa = 0;
        this.nome = '';
        this.descricao = '';
        this.dataInicio = new Date();
        this.dataEntrega = new Date();
        this.statusTarefa = StatusTarefa.Pendente;
        this.idAdministrador = 0;
        this.favorita = false;
        this.notifyAdmin = false;
        this.observacao = '';

    // inicializar metadados de documento
    this.documentoNome = '';
    this.documentoMime = '';
    this.documentoTamanho = 0;
    this.documentoPath = '';

         //apagar esse usuarios e descobrir a solução
        this.usuarios = [];
        this.usuariosIds = [];
    }

     // Método para resetar dataInicio para agora
    setDataInicioAgora() {
        this.dataInicio = new Date();
    }

    // Método para formatar data para input datetime-local
    getDataEntregaFormatada(): string {
        if (!this.dataEntrega) return '';
        const date = new Date(this.dataEntrega);
        // Formato: YYYY-MM-DDTHH:mm (datetime-local)
        return date.getFullYear() + '-' + 
               String(date.getMonth() + 1).padStart(2, '0') + '-' + 
               String(date.getDate()).padStart(2, '0') + 'T' + 
               String(date.getHours()).padStart(2, '0') + ':' + 
               String(date.getMinutes()).padStart(2, '0');
    }

    // Método para definir data de entrega a partir do input
    setDataEntregaFromInput(dateString: string) {
        if (dateString) {
            this.dataEntrega = new Date(dateString);
        }
    }
}

File: VertaskAPI/src/main/java/br/cefetmg/vertask/model/Tarefa.java
package br.cefetmg.vertask.model;

import java.time.LocalDateTime;
import java.util.List;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor

public class Tarefa {
    private Long idTarefa;
    private String nome;
    private String descricao;
    private Long idAdministrador; // ID do Administrador que criou a tarefa
    private Integer clienteId; // referencia para cliente.id (pessoa ou empresa)
    private StatusTarefa statusTarefa; // Ex: "Pendente", "Em Progresso", "Concluída"
    private LocalDateTime dataInicio;
    private LocalDateTime dataEntrega;
    private List<Long> usuariosIds; // Lista de IDs dos usuários atribuídos à tarefa
    // Metadados do documento anexado (opcional)
    private String documentoNome;
    private String documentoMime;
    private Long documentoTamanho;
    private String documentoPath;
    // Observação livre registrada quando a tarefa é concluída
    private String observacao;
    // Se true, o administrador receberá notificações por e-mail quando a tarefa for iniciada e concluída
    private boolean notifyAdmin = false;
    // Se true, a tarefa está marcada como favorita/prioridade para o usuário (vem de tarefaUsuario)
    private boolean favorita = false;
    

}
